File: pset6/mario/mario.py

#prompt for height
#height range
height_range = [i for i in range(24)]

def inputNumber(message):
  while True:
    try:
       userInput = int(input(message))
    except ValueError:
       print("Please enter an integer")
       continue
    else:
       return userInput
       break


height = inputNumber("Please enter an integer: ")
while height not in height_range:
    height = inputNumber("Please enter an integer: ")
i = 0
space = " "
hash_ = "#"
for i in range(0, height):
    #number of spaces
    n = height - (i + 1)
    #number of hashes
    p = i + 2
    print("{:s}{:s}".format(space * n, hash_ * p))

File: pset1/mario/less/mario.c
# include <cs50.h>
# include <stdio.h>

int main(void)
{
    int height;
    // Prompt the user for input
    do
    {
        height = get_int("Please enter an interger: ");
    }
    while (height < 0 || height > 23);
    for (int i = 0; i < height ; i++)
    {
        // Initialiaze n, number of spaces
        int n = height -(i + 1);
        // Initialiaze p, number of hashes
        int p = i + 2;

        // Print out number of spaces
        for (int j = 0; j < n; j++ )
        {
            printf(" ");
        }
        //Print out number of hashes
        for (int k = 0; k < p; k++)
        {
            printf("#");
        }
        printf("\n");
    }

}


